src/main/java/org/apache/cassandra/sidecar/cluster/InstancesConfig.java

```diff
@@ -37,14 +37,16 @@
      *
      * @param id instance's id
      * @return instance meta information
+     * @throws IllegalArgumentException when the instance with {@code id} does not exist
      */
-    InstanceMetadata instanceFromId(final int id);
+    InstanceMetadata instanceFromId(int id) throws IllegalArgumentException;
```

Comment:
Does `NoSuchElementException` fit better?

Reply:
yeah, I think this makes more sense. I will update it